### PR TITLE
2D semi-simplicial sets

### PR DESCRIPTION
**Pull request by epatters**
This provides the basic data structure and interface. Next up should be orientation, chains and cochains, and the boundary and coboundary (exterior derivative) operators.

### CODE REVIEW

File: src/SimplicialSets.jl [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+end
+
+# 2D simplicial sets
+####################
+
+@present SemiSimplexCategory2D <: SemiSimplexCategory1D begin
+  Tri::Ob
+  src2_first::Hom(Tri,E) # ∂₂(2)
+  src2_last::Hom(Tri,E)  # ∂₂(0)
+  tgt2::Hom(Tri,E)       # ∂₂(1)
```
Comment (jpfairbanks) [inline src/SimplicialSets.jl:80]: these names aren't very intuitive. if they are standard, we could provide an ASCII art diagram in this file.

Comment (epatters) [reply src/SimplicialSets.jl:80]: I will add some docs explaining the names.

I had originally called these fields `edge2`, `edge0`, and `edge1`, which is arguably more standard but is even less intuitive!

Comment (jpfairbanks) [reply src/SimplicialSets.jl:80]: Edge 0-2 at least is completely arbitrary so you just have to memorize what it means. The src first src last is like trying to decode a little puzzle where you think it should be intuitive. As ping as we have docs it doesn't really matter.

**Comment by epatters** [comment]
Going to merge this now. See additional docs. Open to changing the field names if we can come up with better ones.

**Pull request merged**